pkg/remote/client.go

```diff
@@ -0,0 +1,84 @@
+package remote
+
+import (
+	"bytes"
+	"context"
+	"crypto/tls"
+	"fmt"
+	"io"
+	"net/http"
+	"net/url"
+	"time"
+
+	"github.com/gogo/protobuf/proto"
+	"github.com/golang/snappy"
+	prompb "go.buf.build/grpc/go/prometheus/prometheus"
+)
+
+type HTTPConfig struct {
+	UserAgent string
+	Timeout   time.Duration
+	TLSConfig tls.Config
+	BasicAuth BasicAuth
+	Headers   http.Header
+}
+
+type BasicAuth struct {
+	User, Password string
+}
+
+type WriteClient struct {
+	hc  *http.Client
+	url *url.URL
+	cfg *HTTPConfig
+}
+
+func NewWriteClient(endpoint string, cfg *HTTPConfig) (*WriteClient, error) {
+	if cfg == nil {
+		cfg = &HTTPConfig{}
+	}
+	u, err := url.Parse(endpoint)
+	if err != nil {
+		return nil, err
+	}
+	return &WriteClient{
+		hc: &http.Client{
+			Timeout: cfg.Timeout,
+		},
+		url: u,
+		cfg: cfg,
+	}, nil
+}
+
+// Store sends a batch of samples to the HTTP endpoint,
+// the request is the proto marshalled and encoded.
+func (c *WriteClient) Store(ctx context.Context, series []*prompb.TimeSeries) error {
+	b, err := proto.Marshal(&prompb.WriteRequest{
+		Timeseries: series,
+	})
+	if err != nil {
+		return fmt.Errorf("encoding series as protobuf write request failed: %w", err)
+	}
+	b = snappy.Encode(nil, b) // TODO: this call can panic
```

Comment:
:thinking: let's maybe put more details here, otherwise not super-useful TODO :smile:

Reply:
This is some from the old code, now you're mentioning I try to directly fix or open an issue for this.

Reply:
I just spotted that we even mention this in the `README` https://github.com/grafana/xk6-output-prometheus-remote/blob/main/README.md?plain=1#L31

Reply:
Checked the code. Maybe we validate before encoding and return an error instead panic :thinking: 

Something like:
```golang
if snappy.MaxEncodedLen(len(b)) < 0 {
    return errors.New("to large to encode")
}
```

Reply:
Yeah, it was exactly what I was going to do

Reply:
Added in the latest fixup